src/main/java/com/amazon/dlic/auth/http/saml/Saml2SettingsProvider.java

```diff
@@ -95,8 +99,8 @@ Saml2Settings get() throws SamlConfigException {
             // TODO allow overriding of IdP metadata?
             settingsBuilder.fromValues(configProperties);
             settingsBuilder.fromValues(new SamlSettingsMap(this.esSettings));
-
-            return settingsBuilder.build();
+            SpecialPermission.check();
+            return AccessController.doPrivileged((PrivilegedAction<Saml2Settings>) () -> settingsBuilder.build());
```

Comment:
This call will wrap all exceptions into PriviledgedActionException. Could you check if this method throws any exception and write code to propagate it correctly.

Example: https://github.com/opendistro-for-elasticsearch/security/blob/d62ac5334609ea92b913caa5e73917efb8b29043/src/main/java/com/amazon/dlic/auth/ldap2/LDAPAuthenticationBackend2.java#L102

Reply:
It only applies to checked exceptions, as `com.onelogin.saml2.settings.SettingsBuilder.build()` does not throw checked exceptions, the unwrapping similar to what is done in LDAPAuthenticationBackend2.java is not necessary.

Reply:
Answer to my question: https://docs.oracle.com/javase/8/docs/technotes/guides/security/doprivileged.html#handling_exceptions